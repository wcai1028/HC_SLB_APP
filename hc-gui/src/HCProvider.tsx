import React from 'react'
import { Store } from 'redux'
import { A10Provider } from 'a10-gui-framework'

export interface IHCPRoviderProps {
  children: JSX.Element
  store?: Store<any>
  initState?: IObject
}

import appReducers from './redux/reducers'
import config from './settings/config'
const middlewares: any[] = []

const HCProvider = (props: IHCPRoviderProps) => {
  const { children, store, initState } = props
  const options: any = {
    initState,
    reducers: appReducers,
    middlewares,
    CONFIG: config,
    children,
  }
  if (store) {
    options.store = store
  }
  return <A10Provider {...options} />
}
export default HCProvider
