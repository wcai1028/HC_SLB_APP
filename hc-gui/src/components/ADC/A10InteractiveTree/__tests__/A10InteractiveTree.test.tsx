import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'
import HCProvider from '../../../../HCProvider'
import { BrowserRouter, Route } from 'react-router-dom'
import { A10InteractiveTree } from '../index'

const renderTestComponent = props => {
  return ReactTestRenderer.create(
    <HCProvider>
      <BrowserRouter>
        <A10InteractiveTree {...props} />
      </BrowserRouter>
    </HCProvider>,
  )
}

test('should render properly with default icon', () => {
  const onCurrentChange = jest.fn()
  const props = {
    width: 100,
    height: 100,
    data: {
      label: 'test1',
      showChildrenAsStraightLine: true,
    }
  }
  const testComponent = renderTestComponent(props)
  expect(testComponent.toJSON()).toMatchSnapshot()
})

test('should render properly with triangle icon', () => {
  const onCurrentChange = jest.fn()
  const props = {
    width: 100,
    height: 100,
    data: {
      label: 'test1',
      showChildrenAsStraightLine: true,
      iconSymbol: 'triangle'
    }
  }
  const testComponent = renderTestComponent(props)
  expect(testComponent.toJSON()).toMatchSnapshot()
})

test('should render properly with circle icon', () => {
  const onCurrentChange = jest.fn()
  const props = {
    width: 100,
    height: 100,
    data: {
      label: 'test1',
      showChildrenAsStraightLine: true,
      iconSymbol: 'circle'
    }
  }
  const testComponent = renderTestComponent(props)
  expect(testComponent.toJSON()).toMatchSnapshot()
})

test('should render properly with circle icon', () => {
  const onCurrentChange = jest.fn()
  const props = {
    width: 100,
    height: 100,
    data: {
      label: 'test1',
      showChildrenAsStraightLine: true,
      iconSymbol: 'circle'
    }
  }
  const testComponent = renderTestComponent(props)
  expect(testComponent.toJSON()).toMatchSnapshot()
})


test('function _onNodeClick', () => {
  const onCurrentChange = jest.fn()
  const props = {
    width: 100,
    height: 100,
    data: {
      label: 'test1',
      showChildrenAsStraightLine: true,
      iconSymbol: 'circle'
    }
  }
  const testComponent = renderTestComponent(props)
  const tree = testComponent.root.find((node: any) => (node.type as any).name === 'A10InteractiveTree')
  tree.instance._onNodeClick({
    data: {
      label: 'test1',
      showChildrenAsStraightLine: true,
      iconSymbol: 'circle',
      children: [{
        label: 'test2',
        showChildrenAsStraightLine: true,
        iconSymbol: 'circle',
      }]
    }
  })
})
