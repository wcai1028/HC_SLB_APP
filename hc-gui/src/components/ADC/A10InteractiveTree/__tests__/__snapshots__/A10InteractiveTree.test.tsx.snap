// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly with circle icon 1`] = `<svg />`;

exports[`should render properly with circle icon 2`] = `<svg />`;

exports[`should render properly with default icon 1`] = `<svg />`;

exports[`should render properly with triangle icon 1`] = `<svg />`;
